flink-python/pyflink/table/environment_settings.py

```diff
@@ -0,0 +1,199 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+from pyflink.java_gateway import get_gateway
+
+__all__ = ['EnvironmentSettings']
+
+
+class EnvironmentSettings(object):
+    """
+    Defines all parameters that initialize a table environment. Those parameters are used only
+    during instantiation of a :class:`~pyflink.table.TableEnvironment` and cannot be changed
+    afterwards.
+
+    Example:
+    ::
+
+        >>> EnvironmentSettings.new_instance() \\
+        ...     .use_old_planner() \\
+        ...     .in_streaming_mode() \\
+        ...     .with_built_in_catalog_name("my_catalog") \\
+        ...     .with_built_in_database_name("my_database") \\
+        ...     .build()
+    """
+
+    class Builder(object):
+        """
+        A builder for :class:`EnvironmentSettingss`.
```

Comment:
typo: EnvironmentSettingss -> EnvironmentSettings. EnvironmentSettingss also appear in other places, should be updated as well.